### PR TITLE
New: newline-before-return rule (fixes #5009)

### LINKED ISSUE #5009: New Rule Suggestion: newline-before-return - continuation
I would like to continue the work started in https://github.com/eslint/eslint/pull/2439
If you guys are still inline with adding the rule, I will make a PR with the requested changes.

Hopefully this could land in 2.0.0? Thoughts?

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/30041829-new-rule-suggestion-newline-before-return-continuation?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

File: docs/rules/newline-before-return.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+This rule aims to increase code clarity by requiring a blank line before `return` statements, except when the `return` is alone inside a statement group (such as an if statement). In the latter case, the `return` statement does not need to be delineated by virtue of it being alone.
+
+When this rule is on, the follow patterns are considered problems:
+
+```js
+/*eslint newline-before-return: 2*/
+
+function foo() {
+
+    return;         // Error: Unexpected newline before return statement.
```
Comment (nzakas) [inline docs/rules/newline-before-return.md]: We are no longer putting comments in showing the error message. Can you remove them?

File: docs/rules/newline-before-return.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+## Rule Details
+
+This rule aims to increase code clarity by requiring a blank line before `return` statements, except when the `return` is alone inside a statement group (such as an if statement). In the latter case, the `return` statement does not need to be delineated by virtue of it being alone.
+
+When this rule is on, the follow patterns are considered problems:
+
+```js
+/*eslint newline-before-return: 2*/
```
Comment (nzakas) [inline docs/rules/newline-before-return.md]: Can you use "error" instead of 2 in these examples?

**Comment by nzakas** [comment]
This looks really good! Just a couple small things on the docs.

**Comment by kaicataldo** [comment]
Thanks! Updated :)

**Comment by nzakas** [comment]
Lgtm. Thanks!

**Pull request merged**